nucypher/blockchain/eth/utils.py

```diff
@@ -136,17 +134,17 @@ def rpc_endpoint_health_check(endpoint: str, max_drift_seconds: int = 60) -> boo
     return True  # finally!
 
 
-def get_default_rpc_endpoints() -> Dict[int, List[str]]:
+@memoize
+def get_default_rpc_endpoints(domain) -> Dict[int, List[str]]:
     """
     Fetches the default RPC endpoints for various chains
     from the nucypher/chainlist repository.
     """
-    LOGGER.debug(
-        f"Fetching default RPC endpoints from remote chainlist {CHAINLIST_URL}"
-    )
+    url = CHAINLIST_URL + domain + ".json"
```

Comment:
Based on prior comment this could then be

```suggestion
    url = CHAINLIST_URL.format(domain=domain)
```